### PR TITLE
Create UC External Location, Schema, and Table Grants based on workspace-wide Azure SPN mount points

### LINKED ISSUE #94: [FEATURE]: Create UC External Location, Schema, and Table Grants based on workspace-wide Azure SPN mount points
### Problem statement

If we have Azure Blob Storage mount points, data is accessible for all clusters and warehouses in the workspace (at least for those people who have access to those warehouses/clusters). We need to migrate that to UC ACLs.

Related issues:
- https://github.com/databrickslabs/ucx/issues/94
- https://github.com/databrickslabs/ucx/issues/887
- https://github.com/databrickslabs/ucx/issues/340
- https://github.com/databrickslabs/ucx/issues/891
- https://github.com/databrickslabs/ucx/issues/862
- https://github.com/databrickslabs/ucx/issues/337

### Proposed Solution

- Get all permissions for all clusters and sql warehouses
- Create relevant permissions to External Locations that those mounts get converted into

### PR DESCRIPTION
**Pull request by pritishpai**
## Changes
Most work already addressed in https://github.com/databrickslabs/ucx/pull/1285

### Linked issues

Closes #94

### Functionality 

- [ ] added relevant user documentation
- [ ] added new CLI command
- [ ] modified existing command: `databricks labs ucx ...`
- [ ] added a new workflow
- [ ] modified existing workflow: `...`
- [ ] added a new table
- [ ] modified existing table: `...`

### Tests

- [ ] manually tested
- [ ] added unit tests
- [ ] added integration tests
- [ ] verified on staging environment (screenshot attached)

### CODE REVIEW

File: src/databricks/labs/ucx/hive_metastore/grants.py [hunk 1]
```diff
@@ -47,6 +50,7 @@ class Grant:
     table: str | None = None
     view: str | None = None
     udf: str | None = None
+    external_location: str | None = None
```
Comment (nfx) [inline src/databricks/labs/ucx/hive_metastore/grants.py:53]: There's no such thing as external location in legacy table acl and hive metastore

**Review by nfx — CHANGES_REQUESTED** [review]

File: src/databricks/labs/ucx/hive_metastore/grants.py (hunk 1 above)
Comment (pritishpai) [reply src/databricks/labs/ucx/hive_metastore/grants.py]: Removed this since we are just granting access to database and table

**Review by JCZuurmond — APPROVED** [review]
lgtm

**Comment by nfx** [comment]
@pritishpai there is an error: TypeError: 'tuple' object is not callable (1.281s)

**Comment by JCZuurmond** [comment]
@pritishpai : Probably solved in #1441

**Review by nfx — APPROVED** [review]
lgtm

**Pull request merged**